dagshub/data_engine/model/query_result.py

```diff
@@ -297,8 +343,8 @@ def get_blob_fields(
         auth = self.datasource.source.repoApi.auth
 
         def _get_blob_fn(dp: Datapoint, field: str, url: str, blob_path: Path):
-            blob_or_path = _get_blob(url, blob_path, auth, cache_on_disk, load_into_memory)
-            if isinstance(blob_or_path, str):
+            blob_or_path = _get_blob(url, blob_path, auth, cache_on_disk, load_into_memory, path_format)
+            if isinstance(blob_or_path, str) and path_format != "str":
```

Comment:
Any reason to use this unreliable error handling instead of raising errors from `_get_blob`?